### PR TITLE
Multi flatMapOptional

### PR DESCRIPTION
**Pull request by danielkec**
```java
// Item "test" is inlined, same as flatMapping Single.just("test")
Single.just(Optional.of("test"))
          .flatMapOptional(Function.identity())        
          .await();
> test

// No item is inlined from empty optional, same as flatmapping Single.empty()
Single.just(Optional.<String>empty())
          .flatMapOptional(Function.identity())
          .defaultIfEmpty(() -> "Nothing here!")
          .await();
> Nothing here!
```
See #3382 for more info

Signed-off-by: Daniel Kec <daniel.kec@oracle.com>

### CODE REVIEW

File: common/reactive/src/test/java/io/helidon/common/reactive/MultiFlatMapOptionalTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ *
+ */
+package io.helidon.common.reactive;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+import org.junit.jupiter.api.Test;
+
+public class MultiFlatMapOptionalTest {
```
Comment (tomas-langer) [inline common/reactive/src/test/java/io/helidon/common/reactive/MultiFlatMapOptionalTest.java:24]: I think this should also have tests when the originating stream is not a stream of optionals, such as:
```
     Multi.range(1, 5)
                .flatMapOptional(i -> (i % 2) == 0 ? Optional.of(i) : Optional.<Integer>empty()))
                .subscribe(subscriber);
```
Which should work as well (and should cut the middle man of converting the actual value to optional).
This is valid for all tests I think

Comment (danielkec) [reply common/reactive/src/test/java/io/helidon/common/reactive/MultiFlatMapOptionalTest.java:24]: Not sure I follow, you mean something like this?
```java
        Multi.range(1, 5)
                .flatMapOptional(Optional::of)
                .subscribe(subscriber);
```

Comment (tomas-langer) [reply common/reactive/src/test/java/io/helidon/common/reactive/MultiFlatMapOptionalTest.java:24]: your `flatMapOptional(Function.identity())` is redundant

Comment (danielkec) [reply common/reactive/src/test/java/io/helidon/common/reactive/MultiFlatMapOptionalTest.java:23]: fixed

**Review by tomas-langer — DISMISSED** [review]
LGTM

**Review by tomas-langer — APPROVED** [review]
LGTM

**Pull request merged**